### DISCUSSION TITLE
Validate according to the proportion of NAs in a column

### DISCUSSION THREAD

**Discussion by feddelegrand7** [rstudio/pointblank#363] [Q&A] [answered]
Hi Richard, 

Thanks for the great package. I'm wondering if there is a function that allows to calculate the proportion of NAs available within a column. For example, if one column has more than 20% of its values as NAs, the validation should fail. 

Thanks again. 
Fodil.

**Comment by rich-iannone** [comment, marked as answer]
@feddelegrand7 for this type of validation you could use the `col_vals_not_null()` function (see https://rich-iannone.github.io/pointblank/reference/col_vals_not_null.html for help and examples). This checks for `NA`s in data frames and `NULL`s in database tables. 

To set the failure level to 20%, use `action_levels(stop_at = 0.2)` for the `actions` argument in `create_agent()`. More help on this is available at https://rich-iannone.github.io/pointblank/reference/action_levels.html. 

Hope this helps!
